### ISSUE TITLE
Add offline support

### ISSUE THREAD

**Issue by momobobe** [KRTirtho/spotube#102]
Current version has ~~totally no~~ little offline ability (error when offline). Please allow to view cached content (home, songs, albums, etc.) when the device goes offline. Maybe before that we should improve the ability of cache (if there is any) since now even contents on home screen are not cached at all, and scrolling up & down would always reload titles and covers, which is definitely a waste of network consumption, especially for mobile devices.

**Comment by KRTirtho** [comment]
Well, the latest version of Spotube in-memory caching is implemented but for offline caching there's not many good package available imo. Riverpod's `FutureProvider` only supports in-memory caching. `flutter_cache_manager` is available but creating custom Cache Models for each request is time consuming but doable

**Comment by KRTirtho** [comment]
@momobobe I wrote `fl-query` package which allows caching to disk. In v2.7.0 I've added persistent caching in some parts of the app (Queue tracks, Favorite Tracks etc)
And, with `fl-query` it's now possible to cache anything, any type of http response. I'll add this caching support in other areas of Spotube too

So I need suggestions about which parts of the app will make more sense to be persisted. I know caching User Library is one of the most wanted one's :smile:
